src/backend/metal/src/command.rs

```diff
@@ -4241,11 +4316,47 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
             .issue(self.state.push_cs_constants(pc));
     }
 
-    unsafe fn execute_commands<'a, T, I>(&mut self, _cmd_buffers: I)
+    unsafe fn execute_commands<'a, T, I>(&mut self, cmd_buffers: I)
     where
         T: 'a + Borrow<CommandBuffer>,
         I: IntoIterator<Item = &'a T>,
     {
-        unimplemented!()
+        for cmd_buffer in cmd_buffers {
+            let outer_borrowed = cmd_buffer.borrow();
+            let inner_borrowed = outer_borrowed.inner.borrow_mut();
+
+            let (exec_journal, is_inheriting) = match inner_borrowed.sink {
+                Some(CommandSink::Deferred { ref journal, is_inheriting, .. }) => {
+                    (journal, is_inheriting)
+                }
+                _ => panic!("Unexpected secondary sink!"),
+            };
+
+            match *self.inner
+                .borrow_mut()
+                .sink()
+            {
+                CommandSink::Immediate { ref mut cmd_buffer, ref mut encoder_state, ref mut num_passes, .. } => {
```

Comment:
If I understand correctly, it looks like `begin` only records SCBs into `Deferred` sinks? If this is the case then when would this branch be used?

Reply:
The SCB itself is always a `CommandSink::Deferred`, but here we are matching on the command buffer that executes SCB externally, and it could be anything.